test/system/src/main/java/io/pravega/test/system/framework/services/PravegaProperties.java

```diff
@@ -0,0 +1,133 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.test.system.framework.services;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * This class encapsulates all system properties to be passed to the operator for a Pravega deployment.
+ * */
+@Slf4j
+@Builder
+public class PravegaProperties {
+    private static final ImmutableMap<String, String> DEFAULT_PROPERTIES = ImmutableMap.<String, String>builder()
+            // Segment store properties
+            .put("autoScale.muteInSeconds", "120")
+            .put("autoScale.cooldownInSeconds", "120")
+            .put("autoScale.cacheExpiryInSeconds", "120")
+            .put("autoScale.cacheCleanUpInSeconds", "120")
+            .put("curator-default-session-timeout", "10000")
+            .put("bookkeeper.bkAckQuorumSize", "3")
+            // Controller properties.
+            .put("controller.transaction.maxLeaseValue", "60000")
+            .put("controller.retention.frequencyMinutes", "2")
+            .put("log.level", "DEBUG")
+            .build();
+
+    private final Map<String, String> properties = new HashMap<String, String>(DEFAULT_PROPERTIES);
+    private PravegaProperties(){}
+
+    /**
+     * This method adds a key only if it is absent, assuming the default values never need to change
+     *
+     * */
+    private void addProperty(String key, String value) {
+        properties.putIfAbsent(key, value);
+    }
+
+    public ImmutableMap<String, String> getProperties() {
+        return ImmutableMap.copyOf(properties);
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder strBuilder = new StringBuilder();
+        for (String e: this.getProperties().keySet()) {
+            strBuilder.append(e);
+            strBuilder.append(":");
+            strBuilder.append(this.getProperties().get(e));
+            strBuilder.append("|");
+        }
+        return strBuilder.toString();
+    }
+
+    /**
+     * This class overrides the lombok builder.
+     * Add all methods for building PravegaProperties (with/without Auth/TLS) to this class
+     */
+    public static final class PravegaPropertiesBuilder {
+        private static final PravegaProperties DEFAULT_INSTANCE = new PravegaProperties();
+
+        public static PravegaProperties buildWithSecurity() {
+            log.debug("Security Builder invoked.");
+            PravegaProperties ctx = new PravegaProperties();
+            addControllerProperties(ctx);
+            addSegmentStoreProperties(ctx);
+            return ctx;
+        }
+
+        public static PravegaProperties build() {
+            log.debug("Default Builder invoked.");
+            return DEFAULT_INSTANCE;
+        }
+
+        private static void addControllerProperties(PravegaProperties ctx) {
+            //auth
+            ctx.addProperty("controller.auth.enabled", "true");
```

Comment:
Should we move the names of the properties to a constants file? That might make it easier to inspect them, as well as modify them as these things change overtime in Pravega.

Reply:
Don't expect names to change too often, as we always need to ensure backward compatibility when changing names. Could be done in the next set of PRs when we have more properties to deal with.

Reply:
Sounds good.